raiden/transfer/channel.py

```diff
@@ -1351,9 +1353,15 @@ def set_settled(channel_state: NettingChannelState, block_number: BlockNumber) -
         channel_state.settle_transaction.result = TransactionExecutionStatus.SUCCESS
 
 
-def update_contract_balance(end_state: NettingChannelEndState, contract_balance: Balance) -> None:
-    if contract_balance > end_state.contract_balance:
+def update_contract_balance(
+    end_state: NettingChannelEndState, contract_balance: Balance, claim: Claim = EMPTY_CLAIM
+) -> None:
+    if (
+        contract_balance > end_state.contract_balance
+        and claim.total_amount > end_state.claim.total_amount
+    ):
```

Comment:
Why do we need this `if`? And if we keep it, shouldn't it be `or` instead of `and`?

Reply:
I assumed that contract balance is what is deposited? If we only allow claims as a deposit then the amounts should always be equal, correct? (Then still the condition is wrong)